Cargo.toml

```diff
@@ -22,3 +22,6 @@ debug = true
 [profile.ethtests]
 inherits = "test"
 opt-level = 3
+
+[patch."https://github.com/privacy-scaling-explorations/halo2.git"]
+halo2_proofs = { git = "https://github.com/scroll-tech/halo2.git", branch = "v1.1" }
```

Comment:
hahah so we need this https://github.com/scroll-tech/poseidon-circuit/issues/28, then we can only use the poseidon primitive crate, no longer needs halo2_proofs, quite confusing.

Reply:
re-export peer dependency is a good pattern

Reply:
i mean we should get rid of halo2_proofs in future.  How to do this?  Split poseidon-circuit crate into 2, one is focusing on hashing computation, not need of halo2, one is focusing on hahsing circuit, need halo2.   Here we only need the former

Reply:
Re-exporting peer dependencies might seem like a shortcut to a clean and concise codebase. After all, it keeps your imports tidy and avoids users from wrestling with dependency conflicts. But this convenience comes at a hidden cost, potentially leading you down a path of dependency management nightmares.
1. Loss of Version Control:  Peer dependencies are meant to be managed by the consumer, not the provider. By re-exporting, you're taking control over a dependency version that another package might need a specific version of. This can lead to conflicts and unexpected behavior when multiple packages using your library have different version requirements for the peer dependency.
Imagine this:  Your library re-exports a peer dependency, say mathjs, at version 5.0.0.  Package A, which uses your library, needs mathjs at version 4.2.1 for specific functionality.  Package B, also using your library, needs the latest mathjs (version 6.0.0).  This creates a conflict.  Package A might break because it relies on features removed in 5.0.0, while Package B might malfunction due to incompatibilities introduced in 6.0.0.
2. Fragile Ecosystem:  Re-exported dependencies become tightly coupled to your library's version. If you update your library and that update requires a newer version of the peer dependency, all your consumers are forced to upgrade as well, even if their code perfectly works with the older version. This creates a domino effect, potentially causing ripple effects throughout your project's ecosystem.
Think of it this way:  You build a house on a foundation designed for a specific weight limit.  By re-exporting a peer dependency as part of your house, you're changing the foundation's specifications.  If you later decide the house needs a stronger foundation, all the houses built on your design (consumers of your library) would need significant work, even if their original design was perfectly functional.
3. The Illusion of Clarity:  While re-exporting might seem to simplify your code on the surface, it actually obscures the true dependency picture. Consumers have no way of knowing which version of the peer dependency they're actually getting or why it's there. This can make debugging issues and understanding code flow more difficult.
Consider this analogy:  You borrow a toolbox from a friend.  They've added their own custom wrench but haven't mentioned it.  When you use the wrench on a project and it breaks something, troubleshooting becomes harder because you weren't aware of the extra tool and its potential impact.

Reply:
> hahahaha so we need this [scroll-tech/poseidon-circuit#28](https://github.com/scroll-tech/poseidon-circuit/issues/28), then we can only use the poseidon primitive crate, no longer needs halo2_proofs, quite confusing.